contracts/interfaces/registries/IIPAssetRegistry.sol

```diff
@@ -36,6 +36,25 @@ interface IIPAssetRegistry is IIPAccountRegistry {
         bytes metadata
     );
 
+    // TODO: replace the IPRegistered event with IPRegisteredPermissionless
+    /// @notice Emits when an IP is officially registered into the protocol.
+    /// @param ipId The canonical identifier for the IP.
+    /// @param chainId The chain identifier of where the IP resides.
+    /// @param tokenContract The token contract address of the IP NFT.
+    /// @param tokenId The token identifier of the IP.
+    /// @param name The name of the IP.
+    /// @param uri The URI of the IP.
+    /// @param registrationDate The date and time the IP was registered.
+    event IPRegisteredPermissionless(
```

Comment:
The explicit "Permissionless" can give a false connotation of the default approach (owner registering IP) as permissioned. Technically they are, but the terms can have unexpected reception among the non-technical people. I think a better name for saying "Permissionless" is "OnBehalf," e.g. how Aave allows callers to deposit or repay on behalf of another address.

Reply:
It is a temporary name, we will rename it to `IPRegistered` after cleaning up "permissioned" register functions.

Reply:
Sounds good. When do we plan to remove the keyword "Permissionless" (in the codebase) and make it just "IPRegistered"?

If it's not a lot of changes, we should do it in this PR. If it's a lot more changes, then we'll just do another PR.